cmd/main.go

```diff
@@ -220,3 +243,71 @@ func initConfigMapTemplates(ctx context.Context, c client.Client, log *zap.Logge
 
 	return nil
 }
+
+// createConfigurationStore initializes and returns a ConfigurationStore instance for the SiteConfig Operator.
+// This function ensures that the necessary configuration ConfigMap exists in the given namespace and
+// retrieves its data. If the ConfigMap does not exist, a default configuration is created.
+func createConfigurationStore(
+	ctx context.Context,
+	client client.Client,
+	namespace string,
+	log *zap.Logger,
+) (*configuration.ConfigurationStore, error) {
+	// Attempt to initialize the SiteConfig Operator configuration
+	data, err := initializeConfiguration(ctx, client, namespace, log)
+	if err != nil {
+		log.Error("Failed to initialize the SiteConfig Operator configuration", zap.Error(err))
+		return nil, err
+	}
+
+	if len(data) == 0 {
+		log.Error("SiteConfig Operator configuration data is empty")
+		return nil, fmt.Errorf("configuration data is empty")
+	}
+
+	// Parse configuration from the retrieved data
+	config := &configuration.Configuration{}
+	if err := config.FromMap(data); err != nil {
+		log.Error("Failed to parse SiteConfig Operator configuration", zap.Error(err))
+		return nil, err
+	}
+
+	return configuration.NewConfigurationStore(config)
+}
+
+// initializeConfiguration creates or retrieves the SiteConfig Operator configuration ConfigMap
+func initializeConfiguration(
+	ctx context.Context,
+	client client.Client,
+	namespace string,
+	log *zap.Logger,
+) (map[string]string, error) {
+	var data map[string]string
+
+	// Retry logic to handle conflicts or transient errors
+	err := retry.RetryOnConflictOrRetriable(k8sretry.DefaultBackoff, func() error {
+		// Attempt to create the default ConfigMap
+		if err := controller.CreateDefaultConfigurationConfigMap(ctx, client, namespace); err != nil {
+			// If already exists, fetch the existing ConfigMap data
+			if errors.IsAlreadyExists(err) {
+				data, err = controller.GetConfigurationData(ctx, client, namespace)
```

Comment:
Nit, but feels like it would be the more common case to get it then create if it's missing

Reply:
Updated!